platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -17,19 +17,24 @@
 package org.gradle.api.problems.internal;
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
+import org.gradle.api.Action;
 import org.gradle.api.problems.ProblemGroup;
 import org.gradle.api.problems.ProblemId;
 import org.gradle.api.problems.Severity;
 import org.gradle.api.problems.SharedProblemGroup;
 
 import javax.annotation.Nullable;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
-public class DefaultProblemBuilder implements InternalProblemBuilder {
+public  class DefaultProblemBuilder implements InternalProblemBuilder {
+
+    private static List<Class<?>> supportedAdditionalDataTypes = ImmutableList.<Class<?>>of(
```

Comment:
🤔 Do you already have plans how to handle extensibility here?

Reply:
For now, I want to keep the `AdditionalData` as sealed class. Gradle should control what are the allowed types and disallow any extensibility from third parties. 
If we have a reasonable use-case for third parties, only then I'd proceed with extensibility. The idea is to do what we do around model building: producers register what types of data they want to expose. Consumers would declare what kind of data they expect and the problems API would only forward the known items. I'd only consider this for spec/implementation if we have a _very_ compelling use-case.